### ISSUE TITLE
LSP auto import not working with neovim-lsp and compe

### ISSUE THREAD

**Issue by gvanbeck** [phpactor/phpactor#1287]
Hey

I installed nvim-lsp confirm the documentation.
For autocompletion I use compe.
The completion works but not the auto-import. Is there some extra configuration to make?
I can confirm that phpactor completion and auto-import works via omnifunc
below is my configuration listed

```
lua << EOF
local nvim_lsp = require('lspconfig')

-- Use an on_attach function to only map the following keys 
-- after the language server attaches to the current buffer
local on_attach = function(client, bufnr)
  local function buf_set_keymap(...) vim.api.nvim_buf_set_keymap(bufnr, ...) end
  local function buf_set_option(...) vim.api.nvim_buf_set_option(bufnr, ...) end

  -- Mappings.
  local opts = { noremap=true, silent=true }

  -- See `:help vim.lsp.*` for documentation on any of the below functions
  buf_set_keymap('n', 'gD', '<Cmd>lua vim.lsp.buf.declaration()<CR>', opts)
  buf_set_keymap('n', 'gd', '<Cmd>lua vim.lsp.buf.definition()<CR>', opts)
  buf_set_keymap('n', 'K', '<Cmd>lua vim.lsp.buf.hover()<CR>', opts)
  buf_set_keymap('n', 'gi', '<cmd>lua vim.lsp.buf.implementation()<CR>', opts)
  buf_set_keymap('n', '<C-k>', '<cmd>lua vim.lsp.buf.signature_help()<CR>', opts)
  buf_set_keymap('n', '<space>wa', '<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>', opts)
  buf_set_keymap('n', '<space>wr', '<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>', opts)
  buf_set_keymap('n', '<space>wl', '<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>', opts)
  buf_set_keymap('n', '<space>D', '<cmd>lua vim.lsp.buf.type_definition()<CR>', opts)
  buf_set_keymap('n', '<space>rn', '<cmd>lua vim.lsp.buf.rename()<CR>', opts)
  buf_set_keymap('n', '<space>ca', '<cmd>lua vim.lsp.buf.code_action()<CR>', opts)
  buf_set_keymap('n', 'gr', '<cmd>lua vim.lsp.buf.references()<CR>', opts)
  buf_set_keymap('n', '<space>e', '<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>', opts)
  buf_set_keymap('n', '[d', '<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>', opts)
  buf_set_keymap('n', ']d', '<cmd>lua vim.lsp.diagnostic.goto_next()<CR>', opts)
  buf_set_keymap('n', '<space>q', '<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>', opts)
  buf_set_keymap("n", "<space>f", "<cmd>lua vim.lsp.buf.formatting()<CR>", opts)

end

-- Use a loop to conveniently call 'setup' on multiple servers and
-- map buffer local keybindings when the language server attaches
local servers = { "phpactor" }
for _, lsp in ipairs(servers) do
  nvim_lsp[lsp].setup { on_attach = on_attach }
end
EOF

lua << EOF
vim.o.completeopt = "menuone,noselect"

require'compe'.setup {
  enabled = true;
  autocomplete = true;
  debug = false;
  min_length = 1;
  preselect = 'enable';
  throttle_time = 80;
  source_timeout = 200;
  incomplete_delay = 400;
  max_abbr_width = 100;
  max_kind_width = 100;
  max_menu_width = 100;
  documentation = true;

  source = {
    path = true;
    buffer = true;
    calc = true;
    nvim_lsp = true;
    nvim_lua = true;
    vsnip = true;
    ultisnips = true;
  };
}
EOF

inoremap <silent><expr> <C-Space> compe#complete()
inoremap <silent><expr> <CR>      compe#confirm('<CR>')
inoremap <silent><expr> <C-e>     compe#close('<C-e>')
inoremap <silent><expr> <C-f>     compe#scroll({ 'delta': +4 })
inoremap <silent><expr> <C-d>     compe#scroll({ 'delta': -4 })
```
Also the listing of my `:imap`

```
i  <C-Space>   * compe#complete()
i  <C-D>       * compe#scroll({ 'delta': -4 })
i  <C-E>       * compe#close('<C-E>')
i  <C-F>       * compe#scroll({ 'delta': +4 })
i  <C-G>S        <Plug>ISurround
i  <C-G>s        <Plug>Isurround
i  <CR>        * compe#confirm('<CR>')
i  <C-S>         <Plug>Isurround
i  <C-T>       * <C-R>=UltiSnips#ExpandSnippetOrJump()<CR>
i  <C-U>       * <C-G>u<C-U>
i  <C-W>       * <C-G>u<C-W>
```
Is there something I can debug?

thx

**Comment by dantleech** [comment]
You need to _confirm_ completion to have actions (like import) performed, see: https://github.com/neoclide/coc.nvim/wiki/Completion-with-sources#limitation-of-cocnvims-completion

**Comment by weeman1337** [comment]
Hi @gvanbeck I am already working on that. At the moment Phpactor sends the import as command alongside the completion result. `compe` doesn't handle it at the moment (https://github.com/hrsh7th/nvim-compe/issues/345).

Additionaly the [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion) suggests using text edits for imports.

Long story short wait for #1248

**Comment by weeman1337** [comment]
@gvanbeck could you check whether it is now working for you using the latest `master`? You may need to run an `composer u` in the phpactor directory.

**Comment by ramonbakker** [comment]
@weeman1337 It works great now for me, thanks!

**Comment by dantleech** [comment]
Closed, but feel free to re-open if that's not the case
